docker/docker.yaml

```diff
@@ -0,0 +1,64 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: docker
+spec:
+  params:
+  - name: image
+    description: Reference of the image docker will produce.
+  - name: builder_image
+    description: The location of the docker builder image.
+    default: docker
+  - name: dockerfile
+    description: Path to the Dockerfile to build.
+    default: ./Dockerfile
+  - name: context
+    description: Path to the directory to use as context.
+    default: .
+  - name: build_extra_args
+    description: Extra parameters passed for the build command when building images.
+    default: ""
+  - name: push_extra_args
+    description: Extra parameters passed for the push command when pushing images.
+    default: ""
+  workspaces:
+  - name: source
+  results:
+  - name: IMAGE_DIGEST
+    description: Digest of the image just built.
+  steps:
+  - name: build
+    image: $(params.builder_image)
+    workingDir: $(workspaces.source.path)
+    script: |
+      docker build \
+        $(params.build_extra_args) \
+        --no-cache \
+        -f $(params.dockerfile) -t $(params.image) $(params.context)
+    volumeMounts:
+      - mountPath: /var/run/
+        name: dind-socket
+  - name: push
+    image: $(params.builder_image)
+    workingDir: $(workspaces.source.path)
+    script: |
+      docker push $(params.build_extra_args) $(params.image)
+    volumeMounts:
+      - mountPath: /var/run/
+        name: dind-socket
+  sidecars:
```

Comment:
I think we should align this with our [dind sidecar example](https://github.com/tektoncd/pipeline/blob/2b274a41cf3ffa73e05b12ea96dc2f5027752661/examples/v1beta1/taskruns/dind-sidecar.yaml#L29) which connects to the sidecar over TCP+TLS, and waits for the cert to be generated, instead of sharing the socket. I don't remember exactly but I think there are newer versions of Docker that don't support the socket anymore.